### ISSUE TITLE
Unable to use generated secured API key in backend

### ISSUE THREAD

**Issue by kylegibson** [algolia/algoliasearch-client-python#43]
I expected to be able to use a secured API key in the backend as well as the frontend. This doesn't appear to work:

This is failing for me:

```python

from algoliasearch import algoliasearch

# generate the secured API key
client = algoliasearch.Client('', '')
secured_key = client.generate_secured_api_key('****', 'filters=...')

# Connect to algolia using the secured API key
client = algoliasearch.Client('****', secured_key)
index = client.init_index('testing')
index.add_object({'foo': 123})
```

This results in the exception: `AlgoliaException: Invalid Application-ID or API key`

**Comment by vvo** [comment]
Hi @kylegibson, can you ensure you are using a search key and not the admin key to generate the secured API key?

Also the syntax to generate the key seems to be using hashes and no more using strings? See https://github.com/algolia/algoliasearch-client-python#security--user-api-keys

Let me know

**Comment by kylegibson** [comment]
Hi @vvo 

I have verified that the key being passed to `generate_secured_api_key` is indeed a search key.

Thanks for the tip about using hashes!

**Comment by kylegibson** [comment]
Any more feedback on this issue? I'd like to be able to test the API keys being used client side.

**Comment by dethi** [comment]
Can you give me some information, like python version, API client version, pip version and maybe your OS?

**Comment by kylegibson** [comment]
Python: 2.7.3
API Client: 1.8.0
pip: 1.3.1
OS: Ubuntu 12.04

**Comment by dethi** [comment]
The generated key is **only** for **search**. Did you try to change this line to perform a search instead?
```python
client.index.add_object({'foo': 123})
```

**Comment by kylegibson** [comment]
I understand that the key is only for search. I failed to explain the rationale behind what I was trying to do. I'm writing test cases against the usage of the secured API key to verify that it is secured properly. 

One of the test cases I wanted to write consisted of using the secured API key to attempt to alter the index. My expectation was for the API to generate some kind of exception indicating that the key had insufficient permissions. This expectation was validated when I would use the (non-secured) API key. However, when I used the generated secured API key, I got a very different and unexpected exception, as mentioned above.

A note about the code included at the top of this issue: It was incomplete, I neglected to include the index initialization. I have that code abstracted, so in the process of simplifying it for inclusion here I forgot to include it. 

Per your suggestion, I decided to try the test again using the secured API key and the search operation. This operation completed successfully. I did not receive any "invalid API key" exception. I also tried the operation mentioned originally. Instead of getting the "invalid API key" error, I am now seeing the following exception, which is what I would have originally expected to see:

> AlgoliaException: Method not allowed with this API key

Thanks for the help everyone. I'm not sure what changed, or how it was fixed, but it seems to be working now.
